packages/back-end/src/controllers/datasources.ts

```diff
@@ -546,14 +536,10 @@ export async function putDataSource(
       req.checkPermissions(permissionLevel, updates.projects);
     }
 
-    // If the connection params changed, re-validate the connection
-    // If the user is just updating the display name, no need to do this
-    if (params) {
```

Comment:
Since this is always passed in, this check is always returning true.

I did explore encrypting the updated integration.params and comparing that encrypted string to the existing datasource.params encrypted string, but it was always different.

I explored specifying an `iv` and `padding` when calling `encryptParams` to get it to be more deterministic, but I was unsuccessful. If you have ideas on what might be happening, please let me know. For now, I just removed the check as it wasn't preventing us from running those code anyway.